profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/LanApiLiveTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jclouds.profitbricks.rest.features;
+
+import com.google.common.base.Predicate;
+import java.util.List;
+import org.apache.jclouds.profitbricks.rest.domain.DataCenter;
+import org.apache.jclouds.profitbricks.rest.domain.Snapshot;
+import org.apache.jclouds.profitbricks.rest.domain.State;
+import org.apache.jclouds.profitbricks.rest.domain.Lan;
+import org.apache.jclouds.profitbricks.rest.internal.BaseProfitBricksLiveTest;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+
+@Test(groups = "live", testName = "LanApiLiveTest")
+public class LanApiLiveTest extends BaseProfitBricksLiveTest {
+   
+   DataCenter dataCenter;
+   Lan testLan;
+   Snapshot testSnapshot;
+  
+   @BeforeClass
+   public void setupTest() {
+      dataCenter = createDataCenter();
+   }
+   
+   @AfterClass(alwaysRun = true)
+   public void teardownTest() {
+      if (dataCenter != null)
+         deleteDataCenter(dataCenter.id());
+   }
+     
+   @Test
+   public void testCreateLan() {
+      assertNotNull(dataCenter);
+            
+      testLan = lanApi().create(
+              Lan.Request.creatingBuilder()
+              .dataCenterId(dataCenter.id())
+              .name("jclouds-lan")
+              .build());
+
+      assertNotNull(testLan);
+      assertEquals(testLan.properties().name(), "jclouds-lan");
+      assertLanAvailable(testLan);
+   }
+   
+
+   @Test(dependsOnMethods = "testCreateLan")
+   public void testGetLan() {
+      Lan lan = lanApi().get(dataCenter.id(), testLan.id());
+
+      assertNotNull(lan);
+      assertEquals(lan.id(), testLan.id());
+   }
+
+   @Test(dependsOnMethods = "testCreateLan")
+   public void testList() {
+      List<Lan> lans = lanApi().list(dataCenter.id());
+
+      assertNotNull(lans);
+      assertFalse(lans.isEmpty());
+      assertEquals(lans.size(), 1);
```

Comment:
A more robust assertion would be to check that the list *contains* the created Lan. Just to avoid race conditions if at some point we create something else in this datacenter.